#! /bin/sh /usr/share/dpatch/dpatch-run
## 05_raise_exception_no_strings.dpatch by Luciano Bello <luciano@debian.org>
##
## DP: Raise exceptions, no strings. More info (#585367)

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' trunk~/core/ui/gtkUi/comparator/diffutil.py trunk/core/ui/gtkUi/comparator/diffutil.py
--- trunk~/core/ui/gtkUi/comparator/diffutil.py	2011-05-25 09:41:27.000000000 -0300
+++ trunk/core/ui/gtkUi/comparator/diffutil.py	2011-05-30 12:18:02.000000000 -0300
@@ -90,7 +90,7 @@
             seq1 = IncrementalSequenceMatcher(None, sequences[1], sequences[2]).get_difference_opcodes()
             self.diffs = seq0, seq1
         else:
-            raise "Bad number of arguments to Differ constructor (%i)" % len(sequences)
+            raise ValueError("Bad number of arguments to Differ constructor (%i)" % len(sequences))
 
     def change_sequence(self, sequence, startidx, sizechange, texts):
         assert sequence in (0,1,2)
@@ -297,7 +297,7 @@
                     yield None
                 diffs[i] = matcher.get_difference_opcodes()
         else:
-            raise "Bad number of arguments to Differ constructor (%i)" % len(sequences)
+            raise ValueError("Bad number of arguments to Differ constructor (%i)" % len(sequences))
         self.diffs = diffs
         self.num_sequences = len(sequences)
         self.seqlength = [0,0,0]
